javalib/src/main/scala/java/util/function/Function.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util.function
+
+import scala.scalajs.js.annotation.JavaDefaultMethod
+
+trait Function[T, R] { self =>
```

Comment:
This trait (or rather it's default methods) seems to be untested.

Reply:
> Also, consider splitting into a commit per thing you are adding. Makes it easier to see what got added just form the history:
> 
> * Add java.util.functions.Function
> * Add java.util.functions.BiFunction
> 

Just to make sure I'm 100% doing what you prefer, different commits, but not necessarily different PRs, right?